core/ArchiveProcessor/Rules.php

```diff
@@ -356,4 +364,29 @@ public static function shouldProcessSegmentsWhenReArchivingReports()
     {
         return Config::getInstance()->General['rearchive_reports_in_past_exclude_segments'] != 1;
     }
+
+    public static function isSegmentPluginArchivingDisabled($pluginName, $siteId = null)
+    {
+
+        $pluginArchivingSetting = GeneralConfig::getConfigValue('disable_archiving_segment_for_plugins', $siteId);
+
+        if (empty($pluginArchivingSetting)) {
+            return false;
+        }
+
+
+        if (is_string($pluginArchivingSetting)) {
+
+            //special case return false
+            if (preg_match('/[\'^£$%&*()}{@#~?><>|=_+¬-]/', $pluginArchivingSetting)) {
+                return false;
+            }
+
+            $pluginArchivingSetting = explode(",", "$pluginArchivingSetting");
+        }
+
+        //case-insensitive search
+        return (bool)preg_grep('/'.$pluginName.'/i',$pluginArchivingSetting);
+
+    }
```

Comment:
To fix the previous mentioned problems I would suggest to use something like this:

```suggestion
    public static function isSegmentPluginArchivingDisabled($pluginName, $siteId = null)
    {
        $pluginArchivingSetting = GeneralConfig::getConfigValue('disable_archiving_segment_for_plugins', $siteId);

        if (empty($pluginArchivingSetting)) {
            return false;
        }

        if (is_string($pluginArchivingSetting)) {
            $pluginArchivingSetting = explode(",", $pluginArchivingSetting);
            $pluginArchivingSetting = array_filter($pluginArchivingSetting, function($plugin){
                return Manager::getInstance()->isValidPluginName($plugin);
            });
        }

        $pluginArchivingSetting = array_map('strtolower', $pluginArchivingSetting);

        return in_array(strtolower($pluginName), $pluginArchivingSetting);
    }
```